packages/vector_graphics_compiler/lib/vector_graphics_compiler.dart

```diff
@@ -26,9 +30,79 @@ Future<VectorInstructions> parse(
   return const PaintDeduplicator().optimize(await parser.parse());
 }
 
+const VectorGraphicsCodec _codec = VectorGraphicsCodec();
+
 void main(List<String> args) async {
```

Comment:
Right now the encoder wants to set IDs, but the compiler does too. Would be better if we deferred to one or the other. IMO having the encoder set IDs is easier since it can maintain its own invariants more easily.

Reply:
The compiler wants to use IDs for optimizer passes. I think it's fine if the encoder knows nothing about compiler IDs.